dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py

```diff
@@ -2616,26 +2621,119 @@ def get_list_for_evalcalculatedvari(self):
         return self.list_for_evalcalculatedvari
 
     ##
-    # prepare the lines that constitues the body for evalJCalculatedVarI
+    # prepare the lines that constitues the body for evalCalculatedVarIAdept
     # @param self : object pointer
     # @return
-    def prepare_for_evaljcalculatedvar(self):
-        self.list_for_evaljcalculatedvari.append("  // not needed\n")
+    def prepare_for_evalcalculatedvariadept(self):
+        trans = Transpose(self.reader.auxiliary_vars_to_address_map, self.reader.derivative_residual_vars + self.reader.assign_residual_vars)
+        ptrn_vars = re.compile(r'x\[(?P<varId>[0-9]+)\]')
+        self.list_for_evalcalculatedvariadept.append("  data->simulationInfo->discreteCall = 1;\n")
+
+        num_ternary = 0
+        index = 0
+        for var in self.reader.list_calculated_vars:
+            body = None
+            if var in self.reader.list_complex_calculated_vars:
+                body = []
+                for line in self.reader.dic_calculated_vars_values[var.get_name()]:
+                    line = mmc_strings_len1(line)
+                    line_tmp = line
+                    line_tmp = transform_atan3_operator(line_tmp)
+                    line_tmp = line_tmp.replace("modelica_real", "adept::adouble")
+                    line_tmp = line_tmp.replace("Greater(", "Greater<adept::adouble>(")
+                    line_tmp = line_tmp.replace("Less(", "Less<adept::adouble>(")
+                    line_tmp = line_tmp.replace("GreaterEq(", "GreaterEq<adept::adouble>(")
+                    line_tmp = line_tmp.replace("LessEq(", "LessEq<adept::adouble>(")
+                    line_tmp = line_tmp.replace("Greater)", "Greater<adept::adouble>)")
+                    line_tmp = line_tmp.replace("Less)", "Less<adept::adouble>)")
+                    line_tmp = line_tmp.replace("GreaterEq)", "GreaterEq<adept::adouble>)")
+                    line_tmp = line_tmp.replace("LessEq)", "LessEq<adept::adouble>)")
```

Comment:
Same remark as above: I imagine we begin to have adept specialization in different places in the Python script. It could be nice to have a common method dealing with the "comparator" conversion to adept syntax

Reply:
Solved